arches/app/models/resource.py

```diff
@@ -189,22 +189,31 @@ def bulk_save(resources):
             resource.tiles = resource.get_flattened_tiles()
             tiles.extend(resource.tiles)
 
-        # need to save the models first before getting the documents for index
+        # need to handle if the bulk load is appending tiles to existing resources/
+        existing_resources = Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources])
+        existing_resources_ids = [existing_resource.resourceinstanceid for existing_resource in existing_resources]
+        resources_to_create = [resource for resource in resources if resource.resourceinstanceid not in existing_resources_ids]
+
         start = time()
-        Resource.objects.bulk_create(resources)
+        Resource.objects.bulk_create(resources_to_create)
         TileModel.objects.bulk_create(tiles)
 
-        print(f"Time to bulk create tiles and resources: {datetime.timedelta(seconds=time() - start)}")
+        logger.info(f"Time to bulk save tiles and resources: {datetime.timedelta(seconds=time() - start)}")
 
         start = time()
-        for resource in resources:
+        for resource in resources_to_create:
             resource.save_edit(edit_type="create")
 
-        resources[0].tiles[0].save_edit(note=f"Bulk created: {len(tiles)} for {len(resources)} resources.", edit_type="bulk_create")
+        try:
```

Comment:
Just out of curiosity what prompted you to put this into a try/except?

Reply:
I was not happy with blindly calling `save_edit` on a hardcoded array without null testing.  In .NET I would use `resources[0]?.tiles[0]?.save_edit()` which tests for an object before trying to access an attribute or function. If it did crash out then it would miss the indexing stage.

I'm wondering if the resource save_edit a note should be included to state that it is a bulk operation. Is there any inherent reason why the bulk_create edit record is on a single tile?